### PR TITLE
Add APPLICATIONINSIGHTS_CONFIGURATION_CONTENT env var

### PR DESCRIPTION
**Pull request by trask**
for attach

### CODE REVIEW

File: agent/agent-tooling/src/main/java/com/microsoft/applicationinsights/agent/internal/wasbootstrap/configuration/ConfigurationBuilder.java [hunk 1]
```diff
@@ -56,7 +54,6 @@
     private static final String APPLICATIONINSIGHTS_ROLE_NAME = "APPLICATIONINSIGHTS_ROLE_NAME";
     private static final String APPLICATIONINSIGHTS_ROLE_INSTANCE = "APPLICATIONINSIGHTS_ROLE_INSTANCE";
 
-    private static final String APPLICATIONINSIGHTS_JMX_METRICS = "APPLICATIONINSIGHTS_JMX_METRICS";
```
Comment (heyams) [inline agent/agent-tooling/src/main/java/com/microsoft/applicationinsights/agent/internal/wasbootstrap/configuration/ConfigurationBuilder.java:59]: we no longer support jmx env var?

File: agent/agent-tooling/src/main/java/com/microsoft/applicationinsights/agent/internal/wasbootstrap/configuration/ConfigurationBuilder.java [hunk 2]
```diff
@@ -37,16 +36,15 @@
...
 import com.squareup.moshi.Moshi;
-import com.squareup.moshi.Types;
 import okio.Buffer;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 
 public class ConfigurationBuilder {
 
     private static final String APPLICATIONINSIGHTS_CONFIGURATION_FILE = "APPLICATIONINSIGHTS_CONFIGURATION_FILE";
+    private static final String APPLICATIONINSIGHTS_CONFIGURATION_CONTENT = "APPLICATIONINSIGHTS_CONFIGURATION_CONTENT";
```
Comment (heyams) [inline agent/agent-tooling/src/main/java/com/microsoft/applicationinsights/agent/internal/wasbootstrap/configuration/ConfigurationBuilder.java:47]: is this for accommodating the app service UX work?

File: agent/agent-tooling/src/test/java/com/microsoft/applicationinsights/agent/internal/wasbootstrap/configuration/ConfigurationTest.java [hunk 3]
```diff
@@ -68,6 +69,15 @@ public void shouldParse() throws IOException {
         assertEquals(2, configuration.selfDiagnostics.file.maxHistory);
     }
 
+    @Test
+    public void shouldParseFromEnvVar() throws IOException {
+        envVars.set("APPLICATIONINSIGHTS_CONFIGURATION_CONTENT", "{\"connectionString\": \"InstrumentationKey=55555555-5555-5555-5555-555555555555\"}");
```
Comment (heyams) [inline agent/agent-tooling/src/test/java/com/microsoft/applicationinsights/agent/internal/wasbootstrap/configuration/ConfigurationTest.java:74]: does it support JMX metrics? can we add a test for it?

File: agent/agent-tooling/src/main/java/com/microsoft/applicationinsights/agent/internal/wasbootstrap/configuration/ConfigurationBuilder.java (hunk 1 above)
Comment (trask) [reply agent/agent-tooling/src/main/java/com/microsoft/applicationinsights/agent/internal/wasbootstrap/configuration/ConfigurationBuilder.java]: ya, good point, even though we don't document it anywhere, possible someone found it, will leave for now and plan to deprecate in the future

File: agent/agent-tooling/src/main/java/com/microsoft/applicationinsights/agent/internal/wasbootstrap/configuration/ConfigurationBuilder.java (hunk 2 above)
Comment (trask) [reply agent/agent-tooling/src/main/java/com/microsoft/applicationinsights/agent/internal/wasbootstrap/configuration/ConfigurationBuilder.java:50]: 👍

File: agent/agent-tooling/src/test/java/com/microsoft/applicationinsights/agent/internal/wasbootstrap/configuration/ConfigurationTest.java (hunk 3 above)
Comment (trask) [reply agent/agent-tooling/src/test/java/com/microsoft/applicationinsights/agent/internal/wasbootstrap/configuration/ConfigurationTest.java]: added 👍

**Review by heyams — APPROVED** [review]
👍

**Comment by ramthi** [comment]
We discussed about the max size for this env. variable, give the likely max for env variables is around 32K (windows 32K, Linux possibly more).  We agreed that we will keep this to a max of 8K.  RP integration UX will validate this max size.

**Pull request merged**